massa-bootstrap/src/bindings/client.rs

```diff
@@ -75,25 +73,24 @@ impl BootstrapClientBinder {
         &mut self,
         duration: Option<Duration>,
     ) -> Result<BootstrapServerMessage, BootstrapError> {
-        self.duplex.set_read_timeout(duration)?;
-
-        // peek the signature and message len
-        let peek_len = SIGNATURE_SIZE_BYTES + self.size_field_len;
-        let mut peek_buff = vec![0u8; peek_len];
-        // problematic if we can only peek some of it
-        while self.duplex.peek(&mut peek_buff)? < peek_len {
-            // TODO: Backoff spin of some sort
-        }
+        let deadline = duration.map(|d| Instant::now() + d);
 
-        // construct the signature from the peek
-        let sig_array = peek_buff.as_slice()[0..SIGNATURE_SIZE_BYTES]
+        // read the known-len component of the message
+        let known_len = SIGNATURE_SIZE_BYTES + self.size_field_len;
+        let mut known_len_buff = vec![0u8; known_len];
+        // TODO: handle a partial read
+        self.read_exact_timeout(&mut known_len_buff, deadline)
+            .map_err(|(err, _consumed)| err)?;
+
+        // construct the signature
+        let sig_array = known_len_buff.as_slice()[0..SIGNATURE_SIZE_BYTES]
```

Comment:
see conversation here: https://github.com/massalabs/massa/issues/3842 for motivation behind reverting the use of peek. Original intent was to make the very notion of needing to handle a partial read, as far as the stream state is concerned, as obsolete. If this approach _could_ be valuable, using it now is very premature